src/vm/methodtablebuilder.cpp

```diff
@@ -1520,15 +1520,13 @@ MethodTableBuilder::BuildMethodTableThrowing(
         {
 #if defined(CROSSGEN_COMPILE)
 #if defined(_TARGET_X86_) || defined(_TARGET_AMD64_)
-            if ((!IsNgenPDBCompilationProcess()
+            if (!IsNgenPDBCompilationProcess()
                 && GetAppDomain()->ToCompilationDomain()->GetTargetModule() != g_pObjectClass->GetModule())
-                || (strcmp(className, "Sse") != 0 && strcmp(className, "Sse2") != 0))
 #endif // defined(_TARGET_X86_) || defined(_TARGET_AMD64_)
             {
                 // Disable AOT compiling for managed implementation of hardware intrinsics.
                 // We specially treat them here to ensure correct ISA features are set during compilation
-                // The only exception to this rule are SSE and SSE2 intrinsics in CoreLib - we can
-                // safely expand those because we require them to be always available.
+                // The only exception to this rule are intrinsics in CoreLib.
```

Comment:
What is the behavior if you run crossgen on a machine with only AVX support, but not AVX2 support (as an example)?

Reply:
The architecture of the CPU that crossgen is running on shouldn't matter.

Reply:
This is because of the explicit `m_compilerFlags.Set` calls we do?

Is there any reason why explicitly setting the compiler flags is better than updating the VM logic which sets the CPUID check results?

Reply:
Are you asking why we don't set this up in `EEJitManager::SetCpuInfo`? `EEJitManager` is ifdeffed out in crossgen. This seems to be an existing convention:

https://github.com/dotnet/coreclr/blob/1ab8b90ca9f8748776b00cc21cd5740b53dec67b/src/vm/codeman.cpp#L1280-L1281

Reply:
Ah, I see 👍